src/app/components/Routes/index.js

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect } from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+
+import * as Routes from '@constants/routes';
+import { useGlobalValue } from '@context';
+import { actionsCreator } from '@context/user/actions';
+import { useRequest } from '@hooks';
+import { getUser } from '@services/UserService';
+import { Library, BookDetail, Login } from '@screens';
+import Loader from '@components/Loader';
+
+import { APP_OPTIONS, LOGIN_OPTIONS } from './constants';
+
+const { Navigator, Screen } = createStackNavigator();
+
+function AppRoutes() {
+  const { data, loading } = useRequest(getUser);
+  const [{ userState }, dispatch] = useGlobalValue();
+  const token = userState?.token;
+  const screenOptions = token ? APP_OPTIONS : LOGIN_OPTIONS;
+
+  useEffect(() => {
+    if (data) dispatch(actionsCreator.logIn(data));
+  }, [dispatch, data]);
+
+  if (loading) return <Loader />;
+
+  return (
+    <NavigationContainer>
+      <Navigator screenOptions={screenOptions}>
+        {token ? (
+          <>
+            <Screen
+              name={Routes.LIBRARY.NAME}
+              component={Library}
+              options={{ title: Routes.LIBRARY.TITLE }}
+            />
+            <Screen
+              name={Routes.BOOK_DETAIL.NAME}
+              component={BookDetail}
+              options={{ title: Routes.BOOK_DETAIL.TITLE, hasBackBtn: true }}
+            />
+          </>
+        ) : (
+          <Screen name={Routes.LOGIN.NAME} component={Login} />
+        )}
```

Comment:
does this work? because you are conditional adding only one set of routes depending of the token. We generally add another route that is the InitialLoading and depending the value of the token it navigates to the login or the app

Reply:
Yes, I had my doubts too. But, it does work. Actually the react navigation shows you the same example when working with user authentication. Seems a little bit off to me.  

**Important:** I was planning to do the same as you mention, but the documentation says something that caught my mind -> with this approach the user isn't going to be able to go back to the login if he presses the back button on android & ios devices. Seemed interesting to me. 

> ...and when we press the hardware back button we expect to not be able to go back to the authentication flow. [Source](https://reactnavigation.org/docs/auth-flow)

Reply:
ok then